prompt/prompt-executor/prompt-executor-clients/prompt-executor-bedrock-client/src/jvmMain/kotlin/ai/koog/prompt/executor/clients/bedrock/modelfamilies/amazon/NovaDataModels.kt

```diff
@@ -22,12 +25,74 @@ internal data class NovaMessage(
     val role: String,
     @SerialName("content")
     val content: List<NovaContent>
-)
+) {
+    init {
+        content.forEach {
+            if (it.isToolUse) require(role == "assistant")
+            if (it.isToolResult) require(role == "user")
+        }
+    }
```

Comment:
Here and below:
Do we need that kind of verification in data class ctor? I would expect to see any validation as callsite responsibility, i.e. - DTO should not validate itself.